### PR TITLE
*: filter nil errors

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: close None

Problem Summary: `errors.Collect` will filter nil errors automatically, saving you from the tedious `if err != nil { errs = append(errs, ..) }`. 

What is changed and how it works:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Notable changes

- [ ] Has configuration change
- [ ] Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- [ ] Has weirctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -158,11 +158,9 @@ func (s *SQLServer) onConn(ctx context.Context, conn net.Conn) {
 
 // Close closes the server.
 func (s *SQLServer) Close() error {
-	var errs []error
+	errs := make([]error, 0, 4)
```
Comment (djshow832) [inline pkg/proxy/proxy.go:161]: Why 4?

Comment (xhebox) [reply pkg/proxy/proxy.go:161]: Because there are four managers. So I just give a capacity of four.

Comment (djshow832) [reply pkg/proxy/proxy.go:161]: I'm afraid you misunderstood the comment. I didn't mean https://github.com/pingcap/TiProxy/pull/97/files#diff-104a11712684e520f0affa95e6a053faeb713306a10f69ee99b9729cc9dd1996R208 , I mean here.

Comment (xhebox) [reply pkg/proxy/proxy.go:161]: It is just a guess of possible connections, giving a larger initial start capacity.

**Review by djshow832 — APPROVED** [review]

**Pull request merged**